### PR TITLE
fix: asset account totals

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

fixes fiat and crypto amounts on asset and account pages, see screenshots

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

does not close but related to #2614

## Risk

* fiat or crypto total amounts may be incorrect on asset or individual account pages

### Engineering

* check my logic for filtering by optional account and asset ids, given we use the
  `PortfolioBalancesById` type we can reuse the same generic reduce logic across multiple selectors

### Operations

* follow the same testing steps as #2731, but this applies to the fiat and crypto balances show in
  the screenshots below
* please note the *staked amount* will still be incorrect, this fixes the fiat and crypto totals
  shown
* check that the asset pages aggregates the fiat and crypto amounts across all accounts correctly
* check that the account pages show the individual fiat and crypto amounts correctly

## Screenshots (if applicable)

**after**

*asset* page

image

*account 0* page

image

*account 1* page

image

### CODE REVIEW

File: src/components/AssetHeader/AssetChart.tsx [hunk 1]
```diff
@@ -68,22 +65,12 @@ export const AssetChart = ({ accountId, assetId, isLoaded }: AssetChartProps) =>
   const { price } = marketData || {}
   const assetPrice = toFiat(price) ?? 0
   const [view, setView] = useState(accountId ? View.Balance : View.Price)
-  const accountSpecifier = useAppSelector(state =>
-    selectFirstAccountSpecifierByChainId(state, asset?.chainId),
```
Comment (0xdef1cafe) [inline src/components/AssetHeader/AssetChart.tsx:72]: this was very much obviously wrong, and a duplicate of `accountId`

File: src/state/slices/portfolioSlice/utils.ts [hunk 2]
```diff
@@ -448,3 +453,22 @@ export const isAssetSupportedByWallet = (assetId: AssetId, wallet: HDWallet): bo
...
   }
 }
+
+export const genericBalanceIncludingStakingByFilter = (
+  accountBalances: PortfolioAccountBalancesById,
+  assetId: AssetId | undefined,
+  accountId: AccountId | undefined,
+): string => {
+  const totalByAccountId = Object.entries(accountBalances)
+    .filter(([acctId]) => (accountId ? acctId === accountId : true)) // if no accountId filter, return all
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/utils.ts:463]: the guts of things

File: src/state/slices/portfolioSlice/utils.ts [hunk 3]
```diff
@@ -448,3 +453,22 @@ export const isAssetSupportedByWallet = (assetId: AssetId, wallet: HDWallet): bo
...
+    .reduce<Record<AccountId, BigNumber>>((acc, [accountId, byAssetId]) => {
+      const accountTotal = Object.entries(byAssetId)
+        .filter(([id, _assetBalance]) => (assetId ? id === assetId : true)) // if no assetId filter, return all
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/utils.ts:466]: more guts

**Review by gomesalexandre — APPROVED** [review]
Tested against prod with account 2 with ATOM portfolio balance but no staking data, and account 0 and 1 with both ATOM portfolio balance and staking data.

- **OK** - Crypto / fiat portfolio amounts for account 2 in /accounts page are matching the explorer values:

Prod:

image

This diff:

image

Explorer:

image

- **OK** - Aggregated total is correct in ATOM asset page and is the total of all accounts value in `/accounts/` page 

From an architectural perspective, logic looks sound and will allow to get crypto/fiat balances for all / specific accountId

**Pull request merged**